### ISSUE TITLE
Cuda error 59 : device-side assert triggered

### ISSUE THREAD

**Issue by FlorianRuen** [pytorch/pytorch#31660]
Hello,

I'm using an PyTorch implementation of Yolo for a deep learning model, running on AWS EC2 instance (g3s.xlarge with Tesla M60 GPU).

I have an issue when the training process: 8 classes with around 7000 images (for the moment to estimate the results), batch size = 64 (128 also possible), the error is: **Cuda error 59 : Runtime Error: device-side assert triggered**

I found a lot of subject on the same error, but no one answer really clear to the issue.
- I try to increase the number of images (in case the error is because there is not enought targets)
- I try to decrease the batch size
- I try to change PyTorch and Torchvision versions (using torch==1.0.1.post2 and torchvision==0.2.2.post3)
- I also try to restart all the process on fresh install of my instance

I attach a screenshoot of the error using the CUDA_LAUNCH_BLOCKING=1 to get more accuracy stacktrace, in case it can help

![error_cuda_train](https://user-images.githubusercontent.com/13837778/71527760-10070380-28dd-11ea-903b-f0df75ed1d1b.png)

PS: The error come after the first epoch, before run the test function on images

Do you have any idea to fix this ?

Thanks for your help,

Regards,
Florian

**Comment by recepdagli** [comment]
"Make sure the number of output units match the number of your classes"

https://towardsdatascience.com/cuda-error-device-side-assert-triggered-c6ae1c8fa4c3

**Comment by FlorianRuen** [comment]
@recepdagli I use a config.cfg file for this value, and I generate this file using 8 classes values, so yes this requirement is respected.

I also check all my labels for negative values, and there is no negative values in my labels files.

**Comment by mruberry** [comment]
Thanks for reporting this issue!

It's difficult for us to know what's going on without an ability to reproduce it. Is it possible to create a short, self-contained snippet that we can ourselves that shows the issue? Maybe it's just one function with particular tensor sizes, for example, and you can provide a snippet that creates randomly initialized tensors of those sizes and calls the function?

You may also want to try running part of your network on the CPU to identify the issue and possibly get a more informative error message.

**Comment by ezyang** [comment]
You don't mention what version of PyTorch you're running.
